### PR TITLE
Updates to WebServer, ChakraHost and StartupTask

### PR DESCRIPTION
**Pull request by gloveboxes**
Updates to Web Server - improved network reliability 
StartupTask - commented out StartMostRecentScript and Save calls
ChakraHost - changed scripting processing strategy to avoid race conditons

### CODE REVIEW

File: IoTBlockly/ChakraHost/ChakraHost.csproj [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <AssemblyName>ChakraHost</AssemblyName>
     <DefaultLanguage>en-US</DefaultLanguage>
     <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
-    <TargetPlatformVersion>10.0.10586.0</TargetPlatformVersion>
```
Comment (alecont) [inline IoTBlockly/ChakraHost/ChakraHost.csproj:14]: Dave, why change the target platform?

File: IoTBlockly/ChakraHost/ChakraHostAsync.cs [hunk 2]
```diff
@@ -22,20 +23,24 @@ public class ChakraHostAsync
     {
         private JavaScriptSourceContext currentSourceContext = JavaScriptSourceContext.FromIntPtr(IntPtr.Zero);
         private JavaScriptRuntime runtime;
-        private Queue taskQueue = new Queue();
-        private IAsyncAction workItem;
-        public bool running { get; private set; }
+       // private Queue taskQueue = new Queue();
```
Comment (alecont) [inline IoTBlockly/ChakraHost/ChakraHostAsync.cs]: Uhm... I guess it's ok to remove this as we don't use the continuations... but why remove this? did they create problems?

File: IoTBlockly/ChakraHost/ChakraHostAsync.cs [hunk 3]
```diff
@@ -22,20 +23,24 @@ public class ChakraHostAsync
...
+
         public string result { get; private set; }
 
         private readonly ManualResetEventSlim waitEvent = new ManualResetEventSlim(false);
+        private AutoResetEvent newScriptEvent = new AutoResetEvent(false);
+
+        bool initilised = false;
```
Comment (alecont) [inline IoTBlockly/ChakraHost/ChakraHostAsync.cs]: typo

File: IoTBlockly/ChakraHost/ChakraHostAsync.cs [hunk 4]
```diff
@@ -51,7 +56,7 @@ public string init()
             // ES6 Promise callback
             JavaScriptPromiseContinuationCallback promiseContinuationCallback = delegate (JavaScriptValue task, IntPtr callbackState)
             {
-                taskQueue.Enqueue(task);
+               // taskQueue.Enqueue(task);
```
Comment (alecont) [inline IoTBlockly/ChakraHost/ChakraHostAsync.cs]: if we decide to remove the taskQueue, let's just remove these and not just comment out. I wonder if this queue gets populated ever. Can you check?

File: IoTBlockly/ChakraHost/ChakraHostAsync.cs [hunk 5]
```diff
@@ -90,109 +96,69 @@ public string dispose()
             return "";
         }
 
-        public void runScript(string script)
+
+        public void ScriptProcessor()
```
Comment (alecont) [inline IoTBlockly/ChakraHost/ChakraHostAsync.cs:95]: I like the new ScriptProcessor(). Few comments:
- minor: let's use the same "scriptProcessor()" naming convention to be consistent
- where do we call init()? I don't see a call to init() anymore...

File: IoTBlockly/IoTBlocklyBackgroundApp/IoTBlocklyBackgroundApp.csproj [hunk 6]
```diff
@@ -11,7 +11,7 @@
     <AssemblyName>IoTBlocklyBackgroundApp</AssemblyName>
     <DefaultLanguage>en-US</DefaultLanguage>
     <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
-    <TargetPlatformVersion>10.0.10586.0</TargetPlatformVersion>
+    <TargetPlatformVersion>10.0.14393.0</TargetPlatformVersion>
```
Comment (alecont) [inline IoTBlockly/IoTBlocklyBackgroundApp/IoTBlocklyBackgroundApp.csproj:14]: same as above

File: IoTBlockly/IoTBlocklyBackgroundApp/StartupTask.cs [hunk 7]
```diff
@@ -34,7 +34,7 @@ public sealed class StartupTask : IBackgroundTask
             deferral = taskInstance.GetDeferral();
             var publicFolder = await localFolder.GetFolderAsync("public");
 
-            await StartMostRecentScript();
+            // await StartMostRecentScript();  // problematic if JS Corrupt when stored
```
Comment (alecont) [inline IoTBlockly/IoTBlocklyBackgroundApp/StartupTask.cs:37]: Let's add a REVIEW comment to improve this and that is ok to comment this out

File: IoTBlockly/SimpleWebServer/SimpleWebServer.cs [hunk 8]
```diff
@@ -192,6 +193,7 @@ private Task WriteHeaderNoFlushAsync(Stream resp, int statusCode, string message
                 string header = String.Format(
                     "HTTP/1.1 {0} {1}\r\n" +
                     location +
+                    "Cache-Control:public, max-age=31536000\r\n" + // randon big number to ensure browser gets from cache dglover@microsoft.com
```
Comment (alecont) [inline IoTBlockly/SimpleWebServer/SimpleWebServer.cs]: good change. no need to add your alias to the code. Git history will keep that info

File: IoTBlockly/SimpleWebServer/SimpleWebServer.cs [hunk 9]
```diff
@@ -214,8 +216,8 @@ private Task WriteHeaderNoFlushAsync(Stream resp, int statusCode, string message
         public SimpleWebServer()
         {
             listener = new StreamSocketListener();
-            listener.Control.KeepAlive = true;
-            listener.Control.NoDelay = true;
+            //listener.Control.KeepAlive = true;
```
Comment (alecont) [inline IoTBlockly/SimpleWebServer/SimpleWebServer.cs]: let's just remove these; no need to keep the comments

File: IoTBlockly/SimpleWebServer/SimpleWebServer.cs [hunk 10]
```diff
@@ -246,90 +248,104 @@ public void Post(string path, RouteCallback callback)
...
+                byte[] data = new byte[BufferSize];
+                IBuffer buffer = data.AsBuffer();
+                uint dataRead = BufferSize;
+
+                using (IInputStream input = socket.InputStream)
                 {
-                    await input.ReadAsync(buffer, BufferSize, InputStreamOptions.Partial);
-                    request.Append(Encoding.UTF8.GetString(data, 0, data.Length));
-                    dataRead = buffer.Length;
+                    while (dataRead == BufferSize)
```
Comment (alecont) [inline IoTBlockly/SimpleWebServer/SimpleWebServer.cs]: I like this :-)

File: IoTBlockly/SimpleWebServer/SimpleWebServer.cs [hunk 11]
```diff
@@ -246,90 +248,104 @@ public void Post(string path, RouteCallback callback)
...
-            StringBuilder request = new StringBuilder();
-            byte[] data = new byte[BufferSize];
-            IBuffer buffer = data.AsBuffer();
-            uint dataRead = BufferSize;
-            using (IInputStream input = socket.InputStream)
+            try
             {
-                while (dataRead == BufferSize)
+                // grab the request
+                StringBuilder request = new StringBuilder();
```
Comment (alecont) [inline IoTBlockly/SimpleWebServer/SimpleWebServer.cs:256]: I like this :-)

File: IoTBlockly/SimpleWebServer/SimpleWebServer.cs (hunk 10 above)
Comment (alecont) [inline IoTBlockly/SimpleWebServer/SimpleWebServer.cs:263]: is there a case when dataRead != BufferSize but dataRead > 0? are we leaving some bytes on the ground?

File: IoTBlockly/SimpleWebServer/SimpleWebServer.cs [hunk 12]
```diff
@@ -246,90 +248,104 @@ public void Post(string path, RouteCallback callback)
...
+                            }
                         }
-                    }
-                    if (!handled)
-                    {
-                        foreach (var t in getStaticRoutes)
+                        if (!handled)
                         {
-                            if (path.StartsWith(t.Item1))
+                            foreach (var t in getStaticRoutes)
```
Comment (alecont) [inline IoTBlockly/SimpleWebServer/SimpleWebServer.cs:300]: nothing changed here right? just spaces/tabs?

File: IoTBlockly/SimpleWebServer/SimpleWebServer.csproj [hunk 13]
```diff
@@ -11,7 +11,7 @@
     <AssemblyName>SimpleWebServer</AssemblyName>
     <DefaultLanguage>en-US</DefaultLanguage>
     <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
-    <TargetPlatformVersion>10.0.10586.0</TargetPlatformVersion>
+    <TargetPlatformVersion>10.0.14393.0</TargetPlatformVersion>
```
Comment (alecont) [inline IoTBlockly/SimpleWebServer/SimpleWebServer.csproj:14]: same comment as above

**Comment by alecont** [comment]
Hi Dave, sorry for the delay. I added some comments. can you review?

File: IoTBlockly/ChakraHost/ChakraHost.csproj (hunk 1 above)
Comment (gloveboxes) [reply IoTBlockly/ChakraHost/ChakraHost.csproj:14]: no more reason than the latest SDK was released, I wanted to test it worked ok...

File: IoTBlockly/ChakraHost/ChakraHostAsync.cs (hunk 4 above)
Comment (gloveboxes) [reply IoTBlockly/ChakraHost/ChakraHostAsync.cs]: I did check (a lot) and it's never called

File: IoTBlockly/IoTBlocklyBackgroundApp/IoTBlocklyBackgroundApp.csproj (hunk 6 above)
Comment (gloveboxes) [reply IoTBlockly/IoTBlocklyBackgroundApp/IoTBlocklyBackgroundApp.csproj:14]: as above - more for testing as the 19393 SDK was final release.

File: IoTBlockly/ChakraHost/ChakraHostAsync.cs (hunk 5 above)
Comment (gloveboxes) [reply IoTBlockly/ChakraHost/ChakraHostAsync.cs:95]: Ah capitalised following C# method naming conventions, but am easy:)  init() is called inside the try catch block after the waitone.

File: IoTBlockly/ChakraHost/ChakraHostAsync.cs (hunk 2 above)
Comment (gloveboxes) [reply IoTBlockly/ChakraHost/ChakraHostAsync.cs]: as below, never used.  Commented out as I felt guilty removing your code:)

File: IoTBlockly/ChakraHost/ChakraHostAsync.cs (hunk 4 above)
Comment (gloveboxes) [reply IoTBlockly/ChakraHost/ChakraHostAsync.cs]: will remove - as above felt a little guilty removing your code:)

File: IoTBlockly/SimpleWebServer/SimpleWebServer.cs (hunk 10 above)
Comment (gloveboxes) [inline IoTBlockly/SimpleWebServer/SimpleWebServer.cs]: yes similar thought had crossed my mind - changing to do {} while, testing now

Comment (gloveboxes) [reply IoTBlockly/SimpleWebServer/SimpleWebServer.cs:263]: thought ha crossed my mind too, I look at again

Comment (alecont) [reply IoTBlockly/SimpleWebServer/SimpleWebServer.cs:263]: any news on this? :-)

Comment (alecont) [reply IoTBlockly/SimpleWebServer/SimpleWebServer.cs:263]: checked with Dave offline, we're good here

**Comment by alecont** [comment]
looks good. merging.

**Pull request merged**